### ISSUE TITLE
1000 line diff crashes Atom when trying to stage

### ISSUE THREAD

**Issue by marius311** [atom/github#1425]
I have a Jupyter notebook file with about 1000 (long) lines of text under version control. Any time I click on it in the Atom Git pane after making modifications to it (as if I were to then stage changes), Atom hangs while displaying "File has no contents" in the diff pane, then crashes (with the Force Close / Keep Waiting dialog) several seconds later. I think its related to the size of the file, smaller files work as expected. This wouldn't be so bad as I could just avoid clicking on it, but many times the file gets auto-selected (like when you stage a file right above it, Atom automatically moves one file down). 

On the off chance its anything actually related to the specific file and not just its size, the file is https://github.com/marius311/CMBLensing.jl/blob/master/nb/jointmax_128example/jointmax_128example.ipynb. 

Thanks for any help / suggestions! 

```
$ atom --version
Atom    : 1.26.1
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

```
$ apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.26.1
python 2.7.15rc1
git 2.17.0
```

and Ubuntu 18.04.

**Comment by marius311** [comment]
Many thanks for the detailed attention. I tried your steps above and indeed there was no problem. I think problem is actually with the length of the diff rather than of the file itself. If I replace your step 3 above with the following,

3. Open `nb/jointmax_128example/jointmax_128example.ipynb` in Jupyter and go to `Cell -> All Output -> Clear` then save the file. (this creates a very large diff as all of the plot outputs are deleted). 

then Atom 1.26.1 hangs on step 6, even in safe mode. I have also tried with 1.27.1 and, unlike 1.26.1, after about ~20 second the diff displayed and I could scroll it. Upon trying to close the diff tab though Atom hung again.
